### PR TITLE
fix: types

### PR DESCRIPTION
**Pull request by AxelF44**
I've fixed types ptr of Eventptr and the types of the function return of MouseEvent !

### CODE REVIEW

**Review by flavien-chenu — DISMISSED** [review]
Nickel.

**Review by Yann-Masson — DISMISSED** [review]

**Review by flavien-chenu — DISMISSED** [review]
Nickel.

File: graphics/IWindow.hpp [hunk 1]
```diff
@@ -106,7 +106,7 @@ class shared::graphics::IWindow {
      *
      * @return Icon object of the window
      */
-    virtual const std::string &getIcon(void) const = 0;
+    virtual const std::string getIcon(void) const = 0;
```
Comment (TekMath) [inline graphics/IWindow.hpp]: Remove this useless method

**Review by TekMath — CHANGES_REQUESTED** [review]

**Review by flavien-chenu — APPROVED** [review]
OK.

**Review by TekMath — APPROVED** [review]
good job mate

**Pull request merged**